### PR TITLE
🌱 Use pkgs.k8s.io for apt

### PR DESCRIPTION
**Pull request by chrischdi**
**What this PR does / why we need it**:

Follow-up fix for #2594

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

/test help

### CODE REVIEW

File: test/e2e/data/infrastructure-vsphere/main/install-on-bootstrap/inject-install-on-bootstrap.yaml [hunk 1]
```diff
@@ -61,9 +61,10 @@
           export DEBIAN_FRONTEND=noninteractive
           # sometimes the network is not immediately available, so we have to retry the apt-get update
           retry 10 5 "apt-get update"
-          apt-get install -y apt-transport-https ca-certificates curl
-          curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add -
-          echo 'deb https://apt.kubernetes.io/ kubernetes-xenial main' >/etc/apt/sources.list.d/kubernetes.list
+          apt-get install -y apt-transport-https ca-certificates curl gpg
+          mkdir -p -m 755 /etc/apt/keyrings
+          curl -fsSL https://pkgs.k8s.io/core:/stable:/v1.28/deb/Release.key | gpg --dearmor -o /etc/apt/keyrings/kubernetes-apt-keyring.gpg
```
Comment (sbueringer) [inline test/e2e/data/infrastructure-vsphere/main/install-on-bootstrap/inject-install-on-bootstrap.yaml:66]: Is that 1.28 there versioned by Kubernetes release?

Comment (chrischdi) [reply test/e2e/data/infrastructure-vsphere/main/install-on-bootstrap/inject-install-on-bootstrap.yaml:66]: oh 🤔

**Comment by sbueringer** [comment]
/lgtm
/approve

/hold
Pending CI & when you want to merge

**Comment by sbueringer** [comment]
Thank you very much!

**Comment by sbueringer** [comment]
/override pull-cluster-api-provider-vsphere-e2e-upgrade-1-29-1-30-main

**Pull request merged**